src/App/Pages/Vault/ViewPageViewModel.cs

```diff
@@ -433,6 +457,69 @@ private async void DownloadAttachmentAsync(AttachmentView attachment)
             }
         }
 
+        public async void PromptOpenOrSave(byte[] data, AttachmentView attachment)
+        {
+            var options = new List<KeyValuePair<string, string>>
+            {
+                new KeyValuePair<string, string>("open", AppResources.Open),
+                new KeyValuePair<string, string>("save", AppResources.Save)
+            };
+            var selection = await Page.DisplayActionSheet(attachment.FileName, AppResources.Cancel, null,
+                options.Select(o => o.Value).ToArray());
+            if(selection != null && selection != AppResources.Cancel)
+            {
+                switch(options.FirstOrDefault(f => f.Value == selection).Key)
+                {
+                    case "open":
+                        OpenAttachment(data, attachment);
+                        break;
+
+                    case "save":
+                        SaveAttachment(data, attachment);
+                        break;
+                }
+            }
```

Comment:
I think you are basing this off of one of my more dynamic/complex examples of `DisplayActionSheet`. We can simplify this here. Just pass `AppResources.Open` and `AppResources.Save` and compare that in `selection` result. 

```suggestion
            var selection = await Page.DisplayActionSheet(attachment.FileName, AppResources.Cancel, null,
                AppResources.Open, AppResources.Save);
            if(selection == AppResources.Open)
            {
                OpenAttachment(data, attachment);
            }
            else if(selection == AppResources.Save)
            {
                SaveAttachment(data, attachment);
            }
```

Reply:
Ah, yes that's much better.